lib/v2/screens/profile_screens/profile/components/edit_profile_pic_bottom_sheet/interactor/viewmodels/pick_image_bloc.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:bloc/bloc.dart';
+import 'package:image_cropper/image_cropper.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:seeds/v2/screens/profile_screens/profile/components/edit_profile_pic_bottom_sheet/interactor/viewmodels/bloc.dart';
+
+/// --- BLOC
+class PickImageBloc extends Bloc<PickImageEvent, PickImageState> {
+  PickImageBloc() : super(PickImageState.initial());
+
+  @override
+  Stream<PickImageState> mapEventToState(PickImageEvent event) async* {
+    if (event is GetImage) {
+      try {
+        var image = await ImagePicker().getImage(source: event.source, imageQuality: 50);
+        if (image != null) {
```

Comment:
what if the image is null? Should we show some type of error? Or anything?

Reply:
Yes actually Idk a scenario where it can return null I think it would be a failure at the native level of (IO) of the plugion.

but in that case, as you mentioned in a comment, it would be good to show a generic message to the user who understands that the image or photo could not be obtained. Something like a snackbar with the text _Oops something goes wrong!_

I guess Jason works on the designs of these snacks in the meantime I can leave a TODO comment here to make the change when I have them.

Reply:
**Update**
From the official docs:
```
    final pickedFile = await picker.getImage(source: ImageSource.camera);

    setState(() {
      if (pickedFile != null) {
        _image = File(pickedFile.path);
      } else {
        print('No image selected.');
      }
    });
```
It seems to be the expected behavior, as it happens with the navigations, to return null when an image is not selected, I already did some tests and that is how it works.